homeassistant/components/sensor/linky.py

```diff
@@ -5,88 +5,227 @@
     def update(self):
         """Fetch new state data for the sensor."""
+        _LOGGER.debug("Start of update of linky data")
+        self._lk.update()
+        if not self._lk.success:
+            self._state = STATE_UNAVAILABLE
+            self._attributes = []
+            return
+
+        self._state = self._lk.daily[1][CONSUMPTION]
+        self._attributes["halfhourly"] = [d[CONSUMPTION]
+                                          for d in self._lk.halfhourly]
+        self._attributes["daily"] = [d[CONSUMPTION]
+                                     for d in self._lk.daily]
+        self._attributes["peak_hours"] = sum([
+            d[CONSUMPTION]
+            if any([_between(h[0], h[1], d[TIME])
+                    for h in self._peak_hours])
+            else 0 for d in self._lk.halfhourly]) / 2
+        # From kW for 30 minutes to kWh
+        self._attributes["offpeak_hours"] = sum(
+            [0 if any([_between(h[0], h[1], d[TIME])
+                       for h in self._peak_hours])
+             else d[CONSUMPTION]
+             for d in self._lk.halfhourly]) / 2
+        # From kW for 30 minutes to kWh
+        self._attributes["peak_offpeak_percent"] = ((self._attributes
+                                                     ["peak_hours"] *
+                                                     100) /
+                                                    (self._attributes
+                                                     ["peak_hours"] +
+                                                     self._attributes
+                                                     ["offpeak_hours"]))
+        self._attributes["daily_cost"] = (self._peak_hours_cost *
+                                          self._attributes["peak_hours"] +
+                                          self._offpeak_hours_cost *
+                                          self._attributes["offpeak_hours"])
+        if self._lk.compare_month == 0:
+            self._attributes["monthly_evolution"] = 0
+        else:
+            self._attributes["monthly_evolution"] = (
+                1 - ((self._lk.monthly[0][CONSUMPTION]) /
+                     (self._lk.compare_month))) * 100
+        _LOGGER.debug("Computed values: %s",
+                      str(self._attributes))
+
+
+def _hour_to_min(hour):
+    return sum(map(lambda x, y: int(x)*y, hour.split(":"), [60, 1]))
+
+
+def _between(start, end, hour):
+    min_start = _hour_to_min(start)
+    min_end = _hour_to_min(end)
+    min_hour = _hour_to_min(hour)
+    return (min_start <= min_hour <= min_end
+            if min_start < min_end
+            else min_start >= min_hour >= min_end)
+
+
+class LinkyData:
+    """The class for handling the data retrieval."""
+
+    def __init__(self, username, password, timeout):
+        """Initialize the data object."""
+        self._username = username
+        self._password = password
+        self._timeout = timeout
+        self.client = {}
```

Comment:
Why not create the client here? Regardless it should not be `{}` but rather `None` if it doesn't exist.